### PR TITLE
add how to for creating rpm lockfile

### PR DESCRIPTION
**Pull request by brianwcook**

### CODE REVIEW

**Comment by ralphbean** [comment]
Thanks for this!

A few comments:

* Instead of creating a new doc, I think this should be added as a section to [Using Red Hat activation keys to access subscription content](https://konflux-ci.dev/docs/how-tos/configuring/activation-keys-subscription/). We don't need two separate docs about RH subscription stuff.
* We should link *to* this doc from [the main prefetch docs](https://konflux-ci.dev/docs/how-tos/configuring/prefetching-dependencies/#rpm) and we should link back your new content (once moved to the activation-keys-subscription doc) from the main prefetch docs, so users see how they're related.

They are most likely to discover your new doc content when they are first reading the main prefetch docs and trying it, but failing due to subscription issues.

@gsaslis can I ask you to help review this one please?

File: docs/modules/ROOT/pages/how-tos/configuring/configuring-rpm-prefetch.adoc [hunk 1]
```diff
@@ -0,0 +1,158 @@
+= Configuring rpm lockfile for hermetic builds
+
+The rpm-lockfile-prototype tool uses live dnf metadata to resolve a given rpms.in.yaml file into an rpms.lock.yaml file in which every rpm is exactly specified by location and version. Because if uses live metadata, the configuration of dnf repositories on the system will influence the results.
+
+Let's explore a simple scenario that should illustrate all pertinent parts of the process.
+
+We will createa a lockfile that includes the OpenShift clients rpm `+openshift-clients+` which:
```
Comment (brunoapimentel) [inline docs/modules/ROOT/pages/how-tos/configuring/configuring-rpm-prefetch.adoc:7]:
```suggestion
We will create a lockfile that includes the OpenShift clients rpm `+openshift-clients+` which:
```

File: docs/modules/ROOT/pages/how-tos/configuring/configuring-rpm-prefetch.adoc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+* requires a subscription to the OpenShift product
+* is not locataed in the default Red Hat Enterprise Linux repositories
```
Comment (brunoapimentel) [inline docs/modules/ROOT/pages/how-tos/configuring/configuring-rpm-prefetch.adoc:10]:
```suggestion
* is not located in the default Red Hat Enterprise Linux repositories
```

File: docs/modules/ROOT/pages/how-tos/configuring/configuring-rpm-prefetch.adoc (hunk 1 above)
Comment (brianwcook) [reply docs/modules/ROOT/pages/how-tos/configuring/configuring-rpm-prefetch.adoc]: fixed

**Comment by brianwcook** [comment]
> * Instead of creating a new doc, I think this should be added as a section to [Using Red Hat activation keys to access subscription content](https://konflux-ci.dev/docs/how-tos/configuring/activation-keys-subscription/). We don't need two separate docs about RH subscription stuff.

mmkay I moved it.

> * We should link _to_ this doc from [the main prefetch docs](https://konflux-ci.dev/docs/how-tos/configuring/prefetching-dependencies/#rpm) and we should link back your new content (once moved to the activation-keys-subscription doc) from the main prefetch docs, so users see how they're related.

done

**Review by ralphbean — APPROVED** [review]

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 3]
```diff
@@ -389,6 +389,8 @@ In {ProductName}, from the *Applications* view, select the application build you
 * Select the *Logs* tab.
 * Alternatively, you can click *build-container*. When the right panel opens, select the *Logs* tab to see a partial view of the log for that build.
 
+NOTE: Konflux also supports prefetching RPM content which requires a Red Hat subscription. For more information see xref:./activation-keys-subscription.adoc[Using Red Hat activation keys to access subscription content].
```
Comment (brunoapimentel) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:392]: Nitpick: it'd be slightly better if we also added the right content anchor to the link (I'm pretty sure my following suggestion works, but I haven't tested it)

```suggestion
NOTE: Konflux also supports prefetching RPM content which requires a Red Hat subscription. For more information see xref:./activation-keys-subscription.adoc#hermetic-network-isolated-builds[Using Red Hat activation keys to access subscription content].
```

Comment (brianwcook) [reply docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc]: It worked - added!

**Review by brunoapimentel — APPROVED** [review]

**Pull request merged**